src/sqswatcher/plugins/slurm.py

```diff
@@ -208,5 +209,54 @@ def update_cluster(max_cluster_size, cluster_user, update_events, instance_prope
     return failed, succeeded
 
 
+def _is_node_locked(hostname):
+    try:
+        node_state = get_node_state(hostname)
+        log.info("Node %s currently in state %s", hostname, node_state)
+        for disable_state in SLURM_NODE_DISABLED_STATES:
+            if disable_state in node_state:
+                return True
+    except Exception as e:
+        log.error(
+            "Failed when checking if node is locked with exception %s. Reporting node %s as unlocked.", e, hostname
+        )
+    return False
+
+
+def perform_health_actions(health_events):
+    """Update and write node lists( and gres_nodes if instance has GPU); restart relevant nodes."""
+    failed = []
+    succeeded = []
+    for event in health_events:
+        try:
+            if _is_node_locked(event.host.hostname):
+                log.warning(
+                    "Instance %s/%s currently in disabled state 'draining/drained'. "
```

Comment:
disabled state 'draining/drained'. --> we should use in the message the constant SLURM_NODE_DISABLED_STATES defined before.

Reply:
Done, also moving the message to common.utils